testing/e2e.sh

```diff
@@ -131,7 +131,10 @@ EOF
 
 
 function clear_docker() {
-  docker rmi -f $(docker images -aq) || true
+  # Get the IDs of images except the local registry image "registry:2". We only
```

Comment:
where is this registry:2 coming from?

Reply:
Used in lines 396 and 406 to run a local docker registry to test container push/pull with

Reply:
But here you are just excluding registry:2, and you will still delete images that are not created in the tests iiuc?

Reply:
This is just a minor upgrade from what this code was doing before. Previously it would delete *everything*. With this, I save the registry image which is used in more than one test to avoid having to pull it more than once. It's not perfect but slightly better than before.

Reply:
That sgtm, but we should just be more clear in the comments then.

Reply:
Fixed